src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -266,6 +280,98 @@ def __init__(self, session: "snowflake.snowpark.session.Session") -> None:
+        Alternatively, you can use the `format` method and specify the format, and then use the `load` method to specify the path:
+            df = spark.read.format("csv").option("header", "true").load("path/to/your/file.csv")
+
+        The following example demonstrate how to use a DataFrameReader.
+            >>> # Create a temp stage to run the example code.
+            >>> _ = session.sql("CREATE or REPLACE temp STAGE mystage").collect()
+
+        Example 1:
+            Loading the first two columns of a CSV file and skipping the first header line:
+            >>> from snowflake.snowpark.types import StructType, StructField, IntegerType, StringType
+            >>> _ = session.sql("create file format if not exists csv_format type=csv skip_header=1 null_if='none';").collect()
+            >>> _ = session.file.put("tests/resources/testCSVspecialFormat.csv", "@mystage", auto_compress=False)
+            >>> # Define the schema for the data in the CSV files.
+            >>> schema = StructType([StructField("ID", IntegerType()),StructField("USERNAME", StringType()),StructField("FIRSTNAME", StringType()),StructField("LASTNAME", StringType())])
+            >>> # Create a DataFrame that is configured to load data from the CSV files in the stage with a given schema and format.
+            >>> df = session.read.load("@mystage/testCSVspecialFormat.csv",format="csv",schema=schema,format_name="csv_format")
+            >>> # Load the data into the DataFrame and return an array of rows containing the results.
+            >>> df.collect()
+            [Row(ID=0, USERNAME='admin', FIRSTNAME=None, LASTNAME=None), Row(ID=1, USERNAME='test_user', FIRSTNAME='test', LASTNAME='user')]
+        """
+        if schema:
+            self.schema(schema)
+        if format:
+            self._file_type = format.upper()
+        if not self._file_type:
+            raise SnowparkClientExceptionMessages.DF_MUST_PROVIDE_FILETYPE_FOR_READING_FILE()
+        if self._file_type == "CSV":
+            return self.csv(path, **kwargs)
+        else:
+            return self._read_semi_structured_file(path, self._file_type, **kwargs)
+
+    def format(
+        self, format: Literal["avro", "csv", "json", "xml", "parquet", "orc"]
+    ) -> "DataFrameReader":
+        """
+        Sets the file type for ingestion.
+
+        This method allows specifying the file type for the data to be ingested. The accepted file formats are:
+        - avro
+        - csv
+        - json
+        - xml
+        - parquet
+        - orc
+
+        Alternatives to using the `format` method are:
+        1. Using the `option` method with either "format" or "type" key-value pairs.
+        Example: option("format", "csv") or option("type", "json")
+
+        2. Passing the file type directly into the `load` method.
+        Example: load("path/to/your/file", format="json")
+
+        Note: Ensure that you use a valid file format and use the appropriate method or option to specify it.
+
```

Comment:
```suggestion
        Note:
            Ensure that you use a valid file format and use the appropriate method or option to specify it.
```
You could add a Note using this format. https://www.sphinx-doc.org/en/master/usage/extensions/example_google.html

Reply:
You can also check how the final docstrings look using the readme [here](https://github.com/snowflakedb/snowpark-python/tree/main/docs).